include/mbedtls/x509_csr.h

```diff
@@ -83,6 +83,19 @@ typedef struct mbedtls_x509write_csr {
 }
 mbedtls_x509write_csr;
 
+typedef struct mbedtls_x509_san_node {
```

Comment:
Why not use `mbedtls_x509_subject_alternative_name` for this? It seems quite similar.

Reply:
This will be modified a bit. The type of `name` will be `void*`. The `type` will determine the type of `name`. For simple cases like this this `name` is simple a simple string, but moving forward for more complicated `SubjectAlternativeNames` other types can be used.

Reply:
Actually I think that David has a good point here - it's a good idea to reuse the API as much as possible, and that's the thing we're trying to recreate here, more or less. If `cert_write` and `cert_req` will take care of the user input parsing and transforming it to `mbedtls_x509_subject_alternative_name`, it should be sufficient for our needs.

Reply:
Adapted the code to use `mbedtls_x509_subject_alternative_name` as @davidhorstmann-arm suggested.